src/kOS.Safe/Encapsulation/ScalarValue.cs

```diff
@@ -67,12 +72,6 @@ public static ScalarValue Create(object value)
                 return new ScalarIntValue((int)value);
             }
 
-            var scalarValue = value as ScalarValue;
-            if (scalarValue != null)
-            {
-                return Create(scalarValue.Value);
-            }
-
```

Comment:
I wonder how much savings there was just from moving this alone (previously it wasted time trying all the other types before finding out that no converting needed to be done after all).